src/net_processing.cpp

```diff
@@ -401,7 +401,7 @@ struct CNodeState {
         /* Track when to attempt download of announced transactions (process
          * time in micros -> txid)
          */
-        std::multimap<std::chrono::microseconds, uint256> m_tx_process_time;
+        std::multimap<std::chrono::microseconds, GenTxid> m_tx_process_time;
```

Comment:
Is there a good reason to not use a `CInv` as the value here, rather than creating a new class just for use here?

Reply:
I have a follow-up PR that changes several functions to take a GenTxid instead of is_wtxid/txid_or_wtxid arguments.

Reply:
Added those commits to this PR.